src/components/RestaurantDetailView/index.tsx

```diff
@@ -0,0 +1,18 @@
+import { Component } from 'react';
+import { Restaurant } from '../../types';
+import RestaurantInfo from '../RestaurantInfo';
+import './RestaurantDetailView.css';
+
+type RestaurantDetailViewProps = {
+  restaurant: Restaurant;
+};
+
+export default class RestaurantDetailView extends Component<RestaurantDetailViewProps> {
+  render() {
+    return (
+      <div className="detail-view">
+        <RestaurantInfo restaurant={this.props.restaurant} showLink={true} />
```

Comment:
```suggestion
        <RestaurantInfo restaurant={this.props.restaurant} showLink />
```
이렇게도 가능합니다!

근데 `showLink`는 왜 필요한가요..?
하드코딩 아닐까요?

Reply:
props 를 써주기만 해도 true가 되는군요! 

>근데 showLink는 왜 필요한가요..? 하드코딩 아닐까요?

이와 별개로 `하드코딩이지 않을까?` 라는 코멘트를 이해하지 못했어요 ... ㅠ

``` js
type RestaurantInfoProps = {
  restaurant: Restaurant;
  showLink?: boolean;
};

export default class RestaurantInfo extends Component<RestaurantInfoProps> {
  static defaultProps = {
    showLink: false,
  };

  render() {
    const { category, name, distance, description, link } = this.props.restaurant;

    return (
      <>
        <div className="restaurant__category">
          <img src={categoryIconPath[category]} alt={category} className="category-icon" />
        </div>
        <div className="restaurant__info">
          <h3 className="restaurant__name text-subtitle">{name}</h3>
          <span className="restaurant__distance text-body">캠퍼스부터 {distance}분 내</span>
          <p className="restaurant__description text-body">{description}</p>
        </div>
        {this.props.showLink && (
          <a href={link} target="_blank" rel="noreferrer" className="restaurant__link">
            {link}
          </a>
        )}
      </>
    );
  }
}
```

RestaurantInfo 컴포넌트를 사용하는 2개의 컴포넌트(RestaurantItem, RestaurantDetailView)가 있는데요!

RestaurantItem 컴포넌트에서는 link를 보여주지 않아야하고,
RestaurantDetailView 컴포넌트에서는 link를 보여줘야 했어요...

그래서 link를 쓸지 말지에 대한 props를 받게 되었습니다.

추가 설명 해주시면 감사하겠습니다...!

Reply:
이미 선언된 `Restaurant` 타입을 찾아보면 `Link`라는 타입이 존재합니다.
그럼 거기서 데이터가 있는지 조건 검사를 하면되는데 왜 수동으로 `showLink`라는  조건을 부여하는지에 대한 이야기였어요

Reply:
이미 선언된 `Restaurant` 타입을 찾아보면 `Link`라는 타입이 존재합니다.
그럼 거기서 데이터가 있는지 조건 검사를 하면되는데 왜 수동으로 `showLink`라는  조건을 부여하는지에 대한 이야기였어요

Reply:
요 부분은 제 이야기를 이해하려고 하시면 안됩니다..
로직을 보고 문제점을 캐치하셔야해요